protocols/identify/src/behaviour.rs

```diff
@@ -459,8 +462,11 @@ pub enum Event {
         /// The peer with whom the error originated.
         peer_id: PeerId,
         /// The error that occurred.
-        error: ConnectionHandlerUpgrErr<UpgradeError>,
+        error: libp2p_core::UpgradeError<UpgradeError>,
     },
+
+    /// Timeout expired while attempting to identify the remote.
+    Timeout,
```

Comment:
A timeout is an error so it seems odd that we have an `Error` variant and then this. I'd suggest we either log this internally and don't report it or create a dedicated error enum ourselves.

Reply:
> Does it make sense to you to also do the same for `relay`?

Yes I think we should apply that thinking to all protocols.

Reply:
thanks done, ptal! Also removed what seems to be unrequired `inbound_hop::UpgradeError` and shortcircuit to `inbound_hop::FatalUpgradeError`  see d4672d3796519625dbd888aed80ff92b6389ee0e and see if you agree.